### PR TITLE
Topics/bootstrap

### PR DESCRIPTION
**Pull request by mkubacki**

### CODE REVIEW

File: src/to.frontend/to.frontend/Controllers/HomeController.cs [hunk 1]
```diff
@@ -1,16 +1,88 @@
...
+
+        [HttpPost]
+        [Route("Home/CreateAdmin")]
+        public IActionResult PostCreateAdministrator(CreateAdministratorModel model)
+        {
+            var adminUserExists = _applicationState.Get<bool>("AdminUserExists");
+            if (!adminUserExists)
+            {
+                _applicationState.Set("AdminUserExists", true);
+                _handler.HandleUserCreateRequest(new UserCreateRequest
```
Comment (fatihswyx) [inline src/to.frontend/to.frontend/Controllers/HomeController.cs]: Check status of request method and set application state accordingly

File: src/to.frontend/to.frontend/Factories/IApplicationState.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        void Set<T>(string key, T value);
+    }
+
+    public class ApplicationState : IApplicationState
+    {
+        private readonly Dictionary<string, object> _state;
+
+        public ApplicationState()
+        {
+            _state = new Dictionary<string, object>
```
Comment (fatihswyx) [inline src/to.frontend/to.frontend/Factories/IApplicationState.cs]: check if a concurrent dictionary should be used due to concurrency

File: src/to.frontend/to.frontend/Factories/IApplicationState.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+namespace to.frontend.Factories
+{
+    public interface IApplicationState
+    {
+        T Get<T>(string key);
+        void Set<T>(string key, T value);
+    }
+
+    public class ApplicationState : IApplicationState
```
Comment (fatihswyx) [inline src/to.frontend/to.frontend/Factories/IApplicationState.cs]: move application state implementation/interface to seperate files
check file location ( Factories ? )

File: src/to.frontend/to.frontend/Factories/IApplicationState.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        {
+            _state = new Dictionary<string, object>
+            {
+                {"AdminUserExists", false}
+            };
+        }
+
+        public T Get<T>(string key)
+        {
+            _state.TryGetValue(key, out var result);
```
Comment (fatihswyx) [inline src/to.frontend/to.frontend/Factories/IApplicationState.cs]: check result ?

File: src/to.permissionrepo/to.permissionrepo/PermissionRepo.cs [hunk 5]
```diff
@@ -30,6 +30,65 @@ public PermissionRepo(string rootpath, string fileName)
             return RetrieveRolePermissions(role, dict);
         }
 
+        public Status Initialise()
```
Comment (ufindeisen) [inline src/to.permissionrepo/to.permissionrepo/PermissionRepo.cs]: Mix of integration and logic ...

File: src/to.frontend/to.frontend/Factories/IApplicationState.cs (hunk 3 above)
Comment (gbrzyk) [reply src/to.frontend/to.frontend/Factories/IApplicationState.cs]: Moved to own file and Helpers directory

File: src/to.frontend/to.frontend/Factories/IApplicationState.cs (hunk 2 above)
Comment (gbrzyk) [reply src/to.frontend/to.frontend/Factories/IApplicationState.cs]: Concurrent dictionary is not required at the moment, could be added later on.

File: src/to.frontend/to.frontend/Controllers/HomeController.cs (hunk 1 above)
Comment (gbrzyk) [reply src/to.frontend/to.frontend/Controllers/HomeController.cs]: Status is now checked, user is redirected to error page if administrator user could not be created.

**Pull request merged**